cogs/score.py

```diff
@@ -42,146 +43,217 @@ async def on_ready(self):
+        else:
+            c.execute(
+                "SELECT count(ReacteeID) FROM Reactions WHERE ReacteeID = ? AND ReactionName = ? AND ReacterID != ReacteeID",
+                (m_id, emoji))
+            react_count = c.fetchone()[0]
 
-        for (ID, DisplayName, Upmartlet) in members:
-            table.append((counter, DisplayName, Upmartlet))
-            if counter % 7 == 0 or counter == len(members):
-                table_list.append(
-                    tabulate(table[:counter],
-                             headers=["Rank", "Name", "Score"],
-                             tablefmt="fancy_grid"))
-                del table[:]
-            counter += 1
+        await ctx.send("{}'s {} score is {}.".format(nick, emoji, react_count))
 
-        p = Pages(ctx,
-                  item_list=table_list,
-                  title="Upmartlet ranking",
-                  display_option=(0, 1),
-                  editable_content=False)
-
-        await p.paginate()
+        conn.close()
 
     @commands.command()
-    async def score(self, ctx, member: discord.Member = None):
-        """Displays member's score in upmartlet rankings."""
-
-        member = member if member else ctx.message.author
-        m_id = member.id
-        nick = member.display_name
-
+    async def ranking(self, ctx, emoji: str = None):
+        """Get the ranking for a given emoji. If no arguments are given,
+           displays your total upmartlet/downmartlet score.
+           Arguments: An emoji (optional)
+        """
         conn = sqlite3.connect(self.bot.config.db_path)
         c = conn.cursor()
 
-        c.execute("SELECT Score FROM Members WHERE ID = ?", (m_id, ))
-        score = c.fetchone()
-
-        if not score:
-            score = (0, )
-            c.execute("INSERT INTO Members VALUES (?, ?, ?)", (m_id, nick, 0))
-            conn.commit()
-
-        await ctx.send("{} score is {}.".format(nick, score[0]))
+        if not emoji:
+            c.execute(
+                "SELECT M.Name, count(*) "
+                "FROM Reactions AS R, Members as M "
+                "WHERE R.ReactionName = ? "
+                "AND R.ReacterID != R.ReacteeID "
+                "AND R.ReacteeID = M.ID "
+                "GROUP BY R.ReacteeID "
+                "ORDER BY count(*) DESC", (str(self.UPMARTLET), ))
+            upmartlet_dict = dict(c.fetchall())
+            c.execute(
+                "SELECT M.Name, count(*) "
+                "FROM Reactions AS R, Members as M "
+                "WHERE R.ReactionName = ? "
+                "AND R.ReacterID != R.ReacteeID "
+                "AND R.ReacteeID = M.ID "
+                "GROUP BY R.ReacteeID "
+                "ORDER BY count(*) DESC", (str(self.DOWNMARTLET), ))
+            downmartlet_dict = dict(c.fetchall())
+            total_score_dict = {
+                key: upmartlet_dict.get(key, 0) - downmartlet_dict.get(key, 0)
+                for key in list(upmartlet_dict.keys()) +
```

Comment:
don't need the .keys() here
I would rewrite this as
```python
from itertools import chain
# ...
for key in chain(upmartlet_dict, downmartlet_dict)
```

this will iterate through all the keys without the overhead of putting all the keys into a list first

Reply:
Resolved in 0847cae